cmd/aws.go

```diff
@@ -0,0 +1,253 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime"
+	"sync"
+	"syscall"
+	"time"
+
+	v "github.com/apptio/breakglass/vault"
+	garbler "github.com/michaelbironneau/garbler/lib"
+
+	//"github.com/acidlemon/go-dumper"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/bgentry/speakeasy"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var awsCreateLoginProfile bool
+var awsRole string
+
+type AWSCredentialResp struct {
+	AccessKey     string `mapstructure:"access_key"`
+	SecretKey     string `mapstructure:"secret_key"`
+	SecurityToken string `mapstructure:"security_token"`
+}
+
+// mysqlCmd represents the mysql command
+var awsCmd = &cobra.Command{
+	Use:   "aws",
+	Short: "Get temporary login credentials for aws",
+	Long: `Generates temporary credentials for an AWS account
+and returns a user name and password you can use to login`,
+	Run: func(cmd *cobra.Command, args []string) {
+		debug = viper.GetBool("debug")
+		execConn = viper.GetBool("exec")
+
+		if debug == true {
+			log.SetLevel(log.DebugLevel)
+		}
+
+		if awsRole == "" {
+			log.Fatal("No AWS role host specified. See --help")
+		}
+
+		userName = viper.GetString("username")
+		authMethod = viper.GetString("authmethod")
+		vaultHost = viper.GetString("vault")
+
+		log.WithFields(log.Fields{"username": userName,
+			"authmethod": authMethod,
+			"vaulthost":  vaultHost,
+			"vaultport":  vaultPort}).Debug("mysql host is: ", sshHost)
+
+		if vaultHost == "" {
+			log.Fatal("No Vault host specified. See --help")
+		}
+
+		log.WithFields(log.Fields{"username": userName,
+			"authmethod": authMethod,
+			"vaulthost":  vaultHost,
+			"vaultport":  vaultPort}).Debug("prompting for password")
+
+		userPass, _ = speakeasy.Ask("Please enter your password: ")
+
+		// Get a Vault client
+		client, err := v.New(userName, userPass, authMethod, vaultHost, vaultPort)
```

Comment:
Line 76 through here can be replaced with `client := getVaultClient()` and you wouldn't have to repeat all this code :)